### ISSUE TITLE
Download and Upload progress bar

### ISSUE THREAD

**Issue by leo23co** [microsoft/AzureStorageExplorer#1665]
**Is your feature request related to a problem? Please describe.**
I downloaded a 4Gb json file from a blob storage.  It took about 4 hours, even though the internet connection is quite high (arount 80mbps), which should have downloaded in less than 10 minutes.  In previous attemps, it only downloaded only 800 mb, and there was no message that the download failed.

**Describe the solution you'd like**
There should be a progress bar for uploading and downloading, showing, not only advance percentage but also how many bytes out of the total bytes have been downloaded (in bytes and also in mb or gb according to the most convenience magnitude).  In the end we should be able to see also xxx bytes/xxx bytes downloaded. 

So it should be something like

|||||||||||||||||||||||||---45%----------------| 1.576.892 B/3.504.206 B (1.50MB/3.34MB)

then..

Successfully downloaded 3.504.206 B/3.504.206 B (3.34MB/3.34MB)

Or

Download was incomplete.  1.576.892 B/3.504.206 B (1.50MB/3.34MB) (45%)

**Describe alternatives you've considered**
None.

**Additional context**
I wonder why the download is so slow and unreliable.  Is there any hash verification?

**Comment by MRayermannMSFT** [comment]
Yes. We do a hash check when downloading. Also, we should show a % when doing a download from blob storage. Are you not seeing it?

**Comment by leo23co** [comment]
Thanks for your reply.  You're right about the blob storage.  I do see the progress % and download speed.  I referred to the wrong source, the download I'm referring to is from a Data Lake.  This should also display the same % and speed information as a blob storage download.  I am using AzCopy, but still the download is really slow and unreliable.  

![image](https://user-images.githubusercontent.com/29169607/63360856-b9892780-c334-11e9-9411-fa2fa22339f9.png)


And for both, please also consider to include the progress bar and the downloaded/total bytes count, and also download summary details when completed, including the hash confirmation.

**Comment by hulseware** [comment]
I do a lot of legacy VM conversions into Azure. That means I do a lot of uploads of vhd files into Blob Storage. I have a very hard time with Azure Storage Explorer telling when I am actually done and if they actually completed successfully. It is very important to know what file(s) is/are being uploaded and how far along they are and whether or not they complete. I have had numerous examples of something going wrong, but if you look at the blob storage container, it looks like the file got uploaded because it grabs the full size of the source file when it creates it in the container the second the transfer starts. 

I just have gone through uploading 7 vhd files in three different directories up to two different containers in blob storage. This is the feedback I received from MSAE.

![MSAESampleOutput](https://user-images.githubusercontent.com/9933506/68785475-3b26e380-060c-11ea-94ac-dec9c2ceaf79.JPG)

As you can see, in one case I have individual feedback that the files completed or did not complete. The later files show nothing beyond that they were queued. I don't know if this is a difference between upload files and upload folder or if there is a difference once the AzCopy kicked in (I forgot to set it before I did the first upload). But you can't tell if the redo of the failed file or the other three files I uploaded actually worked or not.

In no case do I ever see a progress indication, which would be extremely helpful in knowing how long I need to leave the process alone before checking on it.

**Comment by jinglouMSFT** [comment]
Queued means that an upload/download task has been put in the job queue waiting for the next available worker thread.  Once a worker thread becomes available, that queued job will get executed and if it succeeded, you will see a green check.  If it failed, you will see a red cross.

**Comment by hulseware** [comment]
Yes, I have discovered that since I wrote that. Something happened to my session and it would queue up things and never process them. Again, not sufficient feedback to tell what’s actually going on – did it even make it to the queue? Was the queue really being processed (it appears that it wasn’t). Since it wasn’t, it would be nice a) to be able to see a message as to why it stopped processing the queue, and b) to be able to nudge it to get it to try to start processing again. There are a lot of user models out there (filezilla, winscp, for instance) that show good feedback during usage. I don’t see why the Storage Explorer doesn’t fully adopt what clearly works with respect to feedback.
